src/v2/view-builder/internals/FormInputFactory.js

```diff
@@ -146,6 +146,27 @@ const createIdpButtons = (remediations) => {
   });
 };
 
+const createPIVButton = (settings, appState) => {
```

Comment:
curious.. how `piv` button is different then `idp` button.. ??

Reply:
i would rather try to merge this code into `createIdpButtons` function.. I know the function will not make sense here. but we can change the function name to be more generalized.. PIV is also identity provider like social idps ?

Reply:
I have some thoughts, but I also noticed that they had different name `RemediationForms.REDIRECT_IDP` and `RemediationForms.PIV_IDP`. so maybe it is ok?

Reply:
For widget these are just buttons, build and rendered based on remediation. It would be great if we just have one function that gives you the list of buttons required in a view..

Reply:
The action of these buttons are different, the remediation form names are different. I think generalizing this too much will result in a lot of if/ elses and be confusing. social idps, piv and custom buttons make more sense in their own functions.

Reply:
I don't see much issues here to merge the duplication. Also this way you don't need to do `unshift` at view level. 
```
const createIdpButtons = (remediations) => {
  const redirectIdpRemediations = remediations.filter(idp => idp.name === RemediationForms.REDIRECT_IDP);
   const pivIdp = appState.get('remediations').filter(idp => idp.name === RemediationForms.PIV_IDP);
    
    let allButtons = redirectIdpRemediations.map(idpObject => {
    .......
    ...
    });
    
    // piv button logic
    allButtons.push(pivButton);
   return allButtons;
  }
```

Reply:
I understand what you're saying, but i still believe it is more readable with piv separated out. It is a `redirect` idp, but it behaves differently in the widget. @haishengwu-okta @mauriciocastillosilva-okta thoughts? I am ok to change this if everyone feels this should be merged. In my opinion it is easier to read and understand when the logic is in smaller pieces. One thing i could do is change `createIdpButtons` to `createSocialIdpButtons`, and then `createIdpButtons` could call `createSocialIdpButtons` and `createPIVButton`. Does that sound ok?

Reply:
`createIdpButtons` is not only `social idps` it also build buttons for any custom configured idps.. like `SAML` etc..